lib/matplotlib/tests/test_legend.py

```diff
@@ -1357,3 +1357,10 @@ def test_loc_validation_string_value():
     ax.legend(loc='upper center')
     with pytest.raises(ValueError, match="'wrong' is not a valid value for"):
         ax.legend(loc='wrong')
+
+
+def test_legend_handle_label_mismatch():
+    pl1 = plt.plot(range(10))
+    pl2 = plt.plot(range(10))
```

Comment:
```suggestion
    pl1, = plt.plot(range(10))
    pl2, = plt.plot(range(10))
```
to store the artists themselves in the variables because `plot()` always returns a list of artists.
Note: The test will work either way, but it would not create the desired language with your code.